tests/page/locator-convenience.spec.ts

```diff
@@ -17,6 +17,37 @@
 
 import { test as it, expect } from './pageTest';
 
+it.describe('picking the best available locator for an existing element', () => {
```

Comment:
How does using a selector which depends on the content of the website such as name of buttons, placeholders more stable than id, data-testid etc.?

Reply:
well, ids are often changing, in some frameworks even for each reload.

data-testids are of course the most stable, but in our use case (see [octomind](www.octomind.dev)) we don't control the actual source of the website we're testing and don't want our customer to have to make changes.

also, but not a very strong argument over testids, if accesibility is of concern to you it of course helps to see if people are breaking the accessible aria-labels etc. if you use them in your tests.